File: MyAPI/MyAPI/Data/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace MyAPI.Data
{
    public class Product
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public double Price { get; set; }
        public string Description { get; set; }
        public int UnitInStock { get; set; }

        [ForeignKey(nameof(Category))]
        public int? CategoryId { get; set; }
        public virtual Category Category { get; set; }

        public string ImgUrl { get; set; }
        public string LastUpdate { get; set; }
        public virtual ICollection<APIUser> FavoritedUsers { get; set; }

        public virtual ICollection<Review> Reviews { get; set; }

        public int Status { get; set; }
        public Product()
        {
            FavoritedUsers = new HashSet<APIUser>();
        }

    }
}

File: MyFE/src/app/class/product.ts
import { Category } from "./category";
import { PromotionInfo } from "./promotion-info";
import { Review } from "./review";
import { Tag } from "./tag";
import { User } from "./user";

export class Product {

    id!:number;
    name!:string;
    price!:number;
    description!:string;
    unitInStock!:number;
    category!:Category;
    imgUrl!:string;
    lastUpdate!:string;
    favoritedUsers!:User[]
    reviews!:Review[]
    promoInfo!:PromotionInfo
    status!:number
    numSales!:number
    tags!:Tag[]

	constructor(id:number,name:string,price:number,des:string,unit:number,cate:string,img:string,last:string) {
        this.id=id;
        this.name=name;
        this.description=des;
        this.price=price
        this.unitInStock=unit;
        this.category.name=cate;
        this.imgUrl=img;
        this.lastUpdate=last;
	}
}
